src/app/components/turnos/gestor-agendas/operaciones-agenda/botones-agenda.component.ts

```diff
@@ -274,6 +274,16 @@ export class BotonesAgendaComponent implements OnInit {
         }
     }
 
+    tienePrestacionIniciada(agenda: IAgenda) {
+        let flag = false;
+        agenda.bloques.forEach(bloque => {
```

Comment:
Se recorren todos los bloques siempre.  Recomiendo el uso de  for of y con un break